deploy/crds/apps.open-cluster-management.io_gitopsclusters.yaml

```diff
@@ -47,6 +43,8 @@ spec:
                 required:
                 - argoNamespace
                 type: object
+              enablePullModel:
+                type: boolean
```

Comment:
@mikeshng @philipwu08 @lennysgarage  when enablePullModel == true, the gitopscluster controller will create a dummy cluster secret and replace the existing valid managed cluster secret in the argocd namespace.  As a result, the push model for the managed cluster will be blocked. 

Considering the skip-reconcile annotation we will add the ArgoCD app, the ArgoCD apps with the annotation will not be handled by ArgoCD application controller. even the managed cluster secret is valid, the ArgoCD app won't be delivered by the push model.

So I am thinking the enablePullModel setting may not be necessary here. But it is OK if we want to remain this feature. So the users will explicitly realize these managed clusters will be used by pull model only.

Reply:
@xiangjingli I think the code @lennysgarage added to union the secret takes care of the issue regarding replacing existing valid managed cluster secret.

Reply:
@mikeshng  Yeah, The union Secret is good enhancement. It combines the labels and annotation together and uses new dummy Secret to override existing server URL. So if enablePullModel is on, the existing valid cluster secret will be replaced with the dummy cluster secret. 

But I am wondering if we still need the enablePullModel flag once the skip-reconcile annotation is merged. Then these ArgoCD apps with the annotation won't be handled by ArgoCD application controller any more.

Reply:
@xiangjingli I remember now. This enablePullModel flag is for the community users to easily import OpenClusterManagement ManagedClusters into ArgoCD clusters. Without this flag, the community user will have to create dummy secrets manually. You are right, this flag is not useful for ACM users. I think we can just leave it and not doc it in ACM.